### ISSUE TITLE
Tweaks for follow active note (as maps do not really "follows" it)

### ISSUE THREAD

**Issue by Darkbluetest** [esm7/obsidian-map-view#92]
Hi

The new search feature is a BIG improvement, and it is well done. Tx. The tags search works perfectly. I wanted to comment on few things, though: 

1- For the "Follow active note" I see that it works as a search, but it does not move / center the map in the note, it just hides all notes and shows the selected one, but if the note is in other part of the map that is outside of the window, you don't see it. The map should move, scalated, and center around the selected note, just as it already does when you search by tag, or whatever.
2- when "follow active note" is activated, it would be fine to have the option to not hide all the rest of notes, just center map and highlight the selected note, for example.
3- The zoom level when searching by a note, is the maximum. Is there an option to set the maximum zoom (when searching)?

For the rest it is much better. Thanks a lot for this new version.

**Comment by esm7** [comment]
I'm happy that you like the new query functionality :)
Regarding your questions:
1. "Follow active note" was supposed to follow the "map follows search results" functionality, to have a consistent behavior. But I see now that it's not working as expected. Will fix that.
2. I indeed want to add to Map View a highlighting ability (e.g. for search results and maybe even for secondary queries). When I do that, "follow active note" can be configured to highlight rather than filter.
3. The search zoom uses the same configuration as "Default zoom for 'show on map' action" in the settings. The default is quite high but you can change that.

**Comment by Darkbluetest** [comment]
Thanks you for your reply. Regarding th point 3 I have tested settings for zoom max and min, and I always get a maximum zoom in, when making a quesry by tagm for exemple. Does it really work on your side? Maybe I am the only one

**Comment by esm7** [comment]
The "map follows search results" issue - fixed in the just-released 2.0.3 version.
Regarding the "default zoom for show on map", can you elaborate more on what you're doing? By the last thing you wrote I get the sense we are talking about different things. Are you referring to the zoom as part of a *search result*, e.g. by choosing a marker or making a search in the search box, or to how markers are zoomed when "map follows search results" is on and the query is updated? (the latter is not configurable right now)

**Comment by Darkbluetest** [comment]
Yes, the zoom now works, it filters out all the other notes and move and center around the selected one. About the "default zoom for show on map" setting, I refer to the zoom level shown in map when I perform a query (or select a note with "follow current note" activated). Even if I have it set to minimum, when I select a note and the map centers around it, or when I make a query, it makes a 100% zoom. I understand that "default zoom for show on map" only applies to right click option, and not to queries.

**Comment by Darkbluetest** [comment]
I have seen that the fix in 2.0.3 is broken in 2.0.4, it does not center the view neither zoom it (it keeps the current zoom level). BTW, in 2.0.3 when it centered and zoomed in, it did a kind of white and quick "flash", instead of make a seamless transition to location and zoom (it is secondary, I know, but somewhat bothering if you switch quickly among notes)

**Comment by esm7** [comment]
Damn, the state system is still too fragile. I will continue to improve it and fix the above.

**Comment by esm7** [comment]
I want to understand something better in your comment from a few days ago.
In 2.0.4 I deliberately removed the transition animations in some cases, to make some state changes faster.
However if I understand you correctly, you're saying that **in 2.0.3 transitions were faster**? If that's the case, in what specific scenario is this the case?

**Comment by Darkbluetest** [comment]
Well in the 2.0.4 there is no transition, or zoom in at all when a new note is open. 

In the 2.0.3 there were transitions, but showed a white flash when the zoom levels were quite different.

**Comment by esm7** [comment]
Should be fixed in 2.0.5, can you try it out?

**Comment by esm7** [comment]
P.S: following your feedback I also added the "default zoom for show on map" setting to be used in the "map follows search results" behavior (as a max zoom).

**Comment by Darkbluetest** [comment]
Hi, yes the 2.0.5 works now as the 2.0.3. That means, that when clicking on a note in note explorer, for example, the map goes in a very fast transition to white then to the position of the new note, all of that through fast fade transitions. Not bad, but could be improved (it helps to see that the map has changed, but it could bother if switching fast between notes).

Regarding the zoom level, it is always stuck at max zoom. The level in "default zoom for show on map" seems to do nothing for the followed notes.

**Comment by Darkbluetest** [comment]
EDIT: No, the zoom level now works, it was my mistake, I forgot to restart Obsidian. Regarding the follow note, it works as described.

**Comment by Darkbluetest** [comment]
Hi, another thing related to this. I am not sure if it is intended. When click in a note in the map, it opens in a new or existing pane but it is automatically added a filter "path" so all the other notes dissppear from the map. Thus we have to click the "x" to clean the query box in order to see the other notes in the map. It make the map navigation slow, maybe I am missing some reason or funtionality but could it be avoided to add the filter when clicking on a note in map without affecting the follow current note behavior? Or at least to put it as optional?

**Comment by esm7** [comment]
I looked into this but the Obsidian API makes it very hacky to implement, leaving lots of room to more bugs, while I'm doing my best to make this mechanism solid so it would stop breaking. So better put it aside for now.
